src/qibo/hamiltonians/hamiltonians.py

```diff
@@ -368,12 +368,12 @@ def __init__(self, form=None, nqubits=None, symbol_map={}, backend=None):
     @property
     def dense(self):
         """Creates the equivalent :class:`qibo.hamiltonians.MatrixHamiltonian`."""
-        if self._dense is None:
-            log.warning(
-                "Calculating the dense form of a symbolic Hamiltonian. "
-                "This operation is memory inefficient."
-            )
-            self.dense = self.calculate_dense()
+        #if self._dense is None:
+        #    log.warning(
+        #        "Calculating the dense form of a symbolic Hamiltonian. "
+        #        "This operation is memory inefficient."
+        #    )
```

Comment:
You can silence warning by using the environment variable `QIBO_LOG_LEVEL`

Reply:
Should we leave it up to the user or should I impose it in the part of the code that raises this warning? It appears whenever I create a diagonal operator created using Pauli-Zs which happens a lot when training it.

Reply:
In a sense it is already left to the user through the `QIBO_LOG_LEVEL` env variable...